exports/dataset.cpp

```diff
@@ -21,6 +21,19 @@
 
 namespace py = pybind11;
 
+template <typename Collection>
+auto getItemBySingleIndex(Collection &self, const std::tuple<Dim, gsl::index> &index) {
+  gsl::index idx{std::get<gsl::index>(index)};
+  auto& dim = std::get<Dim>(index);
```

Comment:
Note that thanks to C++17 we can now write:
```cpp
const auto [dim, idx] = index;
``

Reply:
I need dim as const ref (or simply const), but idx should be non const sn not ref. Smth for this in c++ 17?

Reply:
Not that I know.